src/lib.rs

```diff
@@ -64,31 +93,61 @@ fn parse_effective_cpus(raw_cpus: String) -> Result<c_int> {
 }
 
 fn effective_cpus_count(base_cg: PathBuf) -> Result<c_int> {
-    let mut cur = Some(base_cg.as_path());
+    let raw_cpus = read_cg_controller(base_cg, "cpuset.cpus.effective")?;
+    parse_effective_cpus(raw_cpus)
+}
 
-    while let Some(cg) = cur {
-        let path: PathBuf = [
-            "/sys/fs/cgroup",
-            &cg.to_string_lossy(),
-            "cpuset.cpus.effective",
-        ]
-        .iter()
-        .collect();
-
-        if let Ok(v) = fs::read_to_string(&path) {
-            return parse_effective_cpus(v);
-        }
+fn parse_cfs_quota_as_cpus(cpus_max: String) -> Result<c_int> {
+    let parts: Vec<&str> = cpus_max.split_whitespace().collect();
+    if parts.len() != 2 {
+        return Err(Errno { errno: EINVAL });
+    }
+    let quota: c_int = parts[0].parse().map_err(|_| Errno { errno: EINVAL })?;
+    let period: c_int = parts[1].parse().map_err(|_| Errno { errno: EINVAL })?;
 
-        cur = cg.parent()
+    if period == 0 {
+        return Err(Errno { errno: EINVAL }); // Avoid division by zero
     }
 
-    Err(Errno { errno: ENOENT })
+    Ok((quota / period) as c_int)
```

Comment:
Can you write a little blurb (maybe in the readme or somewhere) about why this is correct? I'd like to have some pseudocode too, so even if people don't use this code directly, they can use it as a reference.